pkg/archive/diff.go

```diff
@@ -81,7 +81,7 @@ func UnpackLayer(dest string, layer io.Reader, options *TarOptions) (size int64,
 			parentPath := filepath.Join(dest, parent)
 
 			if _, err := os.Lstat(parentPath); err != nil && os.IsNotExist(err) {
-				err = system.MkdirAll(parentPath, 0600)
+				err = system.MkdirAll(parentPath, 0755)
```

Comment:
Discussing further with @neersighted @corhere ; we may want to factor out this block, and also apply ownership (same as we do for overlay)), so that ownership is also consistent (may be important for rootless).

Reply:
Updated the ownership. Should be the same as for overlay now.